src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -44,15 +45,45 @@ def shape(self):
         return self.dataset.shape
 
 
-class H5TableDataset(H5Dataset):
+class DatasetOfReferences(with_metaclass(ABCMeta, H5Dataset, ReferenceResolver)):
+
+    @abstractmethod
+    def get_object(self, h5obj):
+        pass
+
+    @classmethod
+    @abstractmethod
+    def get_inverse_class(cls):
+        pass
+
+    def invert(self):
+        if not hasattr(self, '__inverted'):
+            cls = self.get_inverse_class()
+            self.__inverted = call_docval_func(cls.__init__, self.__dict__)
+        return self.__inverted
+
+
+class BuilderResolverMixin(BuilderResolver):
+
+    def get_object(self, h5obj):
+        return self.io.get_builder(h5obj)
+
+
+class ContainerResolverMixin(ContainerResolver):
+
+    def get_object(self, h5obj):
+        return self.io.get_container(h5obj)
+
```

Comment:
Can you clarify why we need these Mixin classes in addition to the ContainerResolver and BuildResolver classes? It looks like the Mixin classes is the only place where these get used.